server/game/GameActions/AttachAction.js

```diff
@@ -48,6 +48,7 @@ class AttachAction extends CardGameAction {
             event.parent.upgrades.push(event.card);
             event.card.parent = event.parent;
             event.card.controller = context.player;
+            event.card.effectiveType = 'upgrade';
```

Comment:
this is problematic, because once you turn the card into an upgrade, it never changes back again (so if you go through your deck, the game won't let you play it as a creature).  Ideally, it's state as an upgrade is tracked by an effect (which will be removed when it leaves play). You could hack it some other way too though

Reply:
Nice catch! I added a test case for this scenario and changed my logic (hacked it another way) in the next push.